tests/testthat/test-tmb-methods.R

```diff
@@ -646,3 +646,49 @@ test_that("response residuals helper function works as expected", {
   expected <- component(object, "y_vector") - as.vector(fitted(object))
   expect_equal(result_rsp, expected)
 })
+
+# simulate.mmrm_tmb ----
+
+test_that("simulate.mmrm returns a data.frame object of correct dimension and with no missing values", {
+  object <- get_mmrm()
+  sims <- simulate(object, nsim = 2)
+  expect_data_frame(sims, any.missing = FALSE, nrows = nrow(object$data), ncols = 2)
+
+  # check that when nsim == 1, we still have a single-column data.frame
```

Comment:
better move comment text into `test_that()` label text, and make a new test if needed for this